openlibrary/plugins/openlibrary/js/ile/utils/SelectionManager/SelectionManager.js

```diff
@@ -82,19 +84,44 @@ export default class SelectionManager {
     /**
      * @param {MouseEvent & { currentTarget: HTMLElement }} clickEvent
      */
-    toggleSelected(clickEvent) {
+    processClick(clickEvent) {
         // If there is text selection or the click is on a link that isn't a select handle, don't do anything
-        if (window.getSelection()?.toString() !== '' ||
+        if ((!clickEvent.shiftKey && window.getSelection()?.toString() !== '') ||
             ($(clickEvent.target).closest('a').is('a') &&
             $(clickEvent.target).not('.ile-select-handle').length > 0)) return;
 
         const el = clickEvent.currentTarget;
+        const elIndex = $(el).closest('li,tr')?.index();
+
+        if (clickEvent.shiftKey && this.lastSelectedIndex !== -1 && Math.abs(elIndex - this.lastSelectedIndex) > 1) {
+            // Shift-Clicking creates unwanted text selections that need to be unset
+            const selection = window.getSelection ? window.getSelection() : document.selection ? document.selection : null;
+            if (!!selection) selection.empty ? selection.empty() : selection.removeAllRanges();
+            const commonParent = $(el).closest('li,tr')?.parent();
+            const allHandles = commonParent.find('.ile-selectable');
+            let affectedHandles;
+            if (elIndex > this.lastSelectedIndex) {
+                affectedHandles = allHandles.slice(this.lastSelectedIndex + 1, elIndex + 1);
+            } else {
+                affectedHandles = allHandles.slice(elIndex, this.lastSelectedIndex);
+            }
+            const stateChange = el.classList.contains('ile-selected') ? 'deselect' : 'select';
```

Comment:
I think this behaviour is a little odd, I think I would expect it to select everything inbetween if there are any unselected in the range. So something like `const hasUnselectedBetween = $(el).prevUntil(this.lastSelected).filter(':not(.ile-selected)')`

Reply:
I think you may have misunderstood how this works. It does select everything in between — I'm just checking each element here rather than passing the work to JQuery (partly because of the sibling issue described above). However, it also needs to handle the opposite case, where you're shift-clicking to _deselect_ a range of elements.

Reply:
https://github.com/internetarchive/openlibrary/assets/6251786/027080a7-4fc9-4a36-8b44-0bde33b8c53d

A few corrections to the video:
- 00:31 "It'll deselect everything" ----> "It'll deselect these two selected items"
- 01:27 "I think it should only deselect if everything I'm shift-clicking over is **already selected**" (not deselected :P)

Reply:
I see what you're getting at now. My logic was to change the state based on the first click, but your preference (and Google's) is to change the state based on the second click.  I can live with that. I've changed the two lines required to switch the behavior.